spec/api/v1_spec.rb

```diff
@@ -0,0 +1,263 @@
+
+      response '200', 'Car added' do
+        schema type: :object,
+               properties: {
+                 name: { type: :string }, model: { type: :string },
+                 price: { type: :integer }, description: { type: :string },
+                 photo: { type: :string }
+               },
+               required: %w[name model price description photo]
+
+        let(:car) do
+          Car.create(
+            name: 'Mercedes', model: 'W-14', photo: 'mercedes-photo.png',
+            price: 12, description: 'Best comfort you can get'
+          )
+        end
+        run_test!
+      end
+
+      response '404', 'Car not added' do
+        let(:car) { 'invalid' }
+        run_test!
+      end
+    end
+  end
+
+  # Delete a particular car
+  path '/cars/{id}' do
+    delete 'Delete a car' do
+      tags 'Cars'
+      produces 'application/json'
+      parameter name: :car_id, in: :path, type: :integer, description: 'Id of the car to delete'
+
+      response '204', 'Car deleted.' do
+        let(:id) { 2 }
+        run_test!
+      end
+
+      response '404', 'Car not found' do
+        let(:id) { 'invalid' }
+        run_test!
+      end
+    end
+  end
+
+  # Reserve a car
+  path '/reservations' do
+    post 'Reserve a car' do
+      tags 'Reserve'
+      produces 'application/json'
+
+      parameter name: :reservation, in: :body, schema: {
+        type: :object, properties: {
+          car_id: { type: :integer }, city: { type: :string }, duration: { type: :integer },
+          reservation_date: { type: :string, format: 'date-time' }, user_id: { type: :integer }
+        }, required: %w[car_id city duration reservation_date user_id]
+      }
+
+      response '201', 'car reserved successfully' do
+        schema type: :object, properties: {
+          car: {
+            type: :object,
+            properties: {
+              id: { type: :integer }, name: { type: :string }, model: { type: :string },
+              photo: { type: :string }, price: { type: :integer }, description: { type: :string }
+            }, required: %w[id name model photo price description reserved]
+          },
+          reservation: {
+            type: :object,
+            properties: {
+              car_id: { type: :integer }, city: { type: :string }, duration: { type: :integer },
+              reservation_date: { type: :string, format: 'date-time' }, user_id: { type: :integer }
+            }, required: %w[car_id city duration reservation_date user_id]
+          }
+        }, required: %w[car reservation]
+
+        let(:reservation) do
+          create(car_id: 1, city: 'San Francisco', duration: 2, reservation_date: '01/04/2023', user_id: 1)
+        end
+
+        run_test!
+      end
+
+      response '404', 'Car is not found' do
+        let(:reservation) do
+          create(car_id: 'string', city: 'San Francisco', duration: 2, reservation_date: '01/04/2023', user_id: 1)
+        end
+
+        run_test!
+      end
```

Comment:
These tests fail on my machine. Is it possible you don't haveenough documentation around these? Double check these tests